registration-processor/post-processor/registration-processor-event-handler-stage/src/main/java/io/mosip/registration/processor/eventhandler/stage/EventHandlerStage.java

```diff
@@ -376,11 +382,51 @@ private CredentialRequestDto getCredentialRequestDto(String regId,String issuerI
 		credentialRequestDto.setEncryptionKey(generatePin());
 		additionalAttributes.put("templateTypeCode",templateTypeCode);
 		additionalAttributes.put("registrationId", regId);
+		getOpencrvsFields(regId, process, issuerId, credentialType, additionalAttributes);
 		credentialRequestDto.setAdditionalData(additionalAttributes);
 
 		return credentialRequestDto;
 	}
 
+	private void getOpencrvsFields(String regId, String process, String issuerId,
+								   String credentialType, Map<String, Object> additionalAttributes) {
+		try {
+			JSONArray jArray=new JSONArray(opencrvsAdditionalParam);
+			for(int i=0;i<jArray.length();i++) {
+				if (process.equalsIgnoreCase(jArray.getJSONObject(i).getString("process"))
+				&& credentialType.equalsIgnoreCase(jArray.getJSONObject(i).getString("credentialType"))
+				&& issuerId.equalsIgnoreCase(jArray.getJSONObject(i).getString("issuer"))) {
+					JSONArray jArray2 = jArray.getJSONObject(i).getJSONArray("fields");
+					for(int j=0;j<jArray2.length();j++) {
+						String additionalAttr = jArray2.get(j).toString();
+						String additionalAttrValue = getCrvsField(regId, process, additionalAttr);
+						additionalAttributes.put(additionalAttr, additionalAttrValue);
+					}
+				}
+			}
+		} catch (JSONException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private String getCrvsField(String regId, String process, String additionalAttr){
+		try {
+			Map<String,String> metaInfo = utilities.getPacketManagerService().getMetaInfo(regId, process, ProviderStageName.EVENT_HANDLER);
```

Comment:
If there are multiple fields configured, we are calling metainfo packetmanager endpoint multiple times. 
We should try to call only once per packet.

Reply:
If a country configures 2 metainfo fields in the properties for a NEW type of packet, then this calls happens twice for processing of every NEW packet right ?